### ISSUE TITLE
feat: The headline is not in the right place.

### ISSUE THREAD

**Issue by YasinSpace** [ZTL-UwU/shadcn-docs-nuxt#64]
### Describe the feature

<img width="150" alt="Screenshot 2024-12-04 at 16 49 41" src="https://github.com/user-attachments/assets/3a129142-9f02-4871-a4d6-413d920f9c1f">

As you can see,

the state in the title is not positioned correctly when we click on a title!

And specify this or add in the feature: Please add this : in config file we can change the size of the sidebar, for example, set it more forward or more backward ( resize ) by default. =)

### Additional information

- [ ] Would you be willing to help implement this feature?

**Comment by Ast3risk-ops** [comment]
Is this the builtin search bar? Usually that's way wider.

**Comment by kalix127** [comment]
I’d like to help fix it, but I could use a bit more info:  
1. Could you share a clearer screenshot of the issue with the title’s state position?  
2. If possible, a quick code snippet of how the titles or sidebar are handled would help me understand the issue better.  

I’ve tried reproducing the issue on my end with the same icon and text, but I didn’t notice any problems. Maybe I’m missing something.

Let me know, and I can start working on this.

**Comment by Ast3risk-ops** [comment]
> I’d like to help fix it, but I could use a bit more info:
> 
>     1. Could you share a clearer screenshot of the issue with the title’s state position?
> 
>     2. If possible, a quick code snippet of how the titles or sidebar are handled would help me understand the issue better.
> 
> 
> I’ve tried reproducing the issue on my end with the same icon and text, but I didn’t notice any problems. Maybe I’m missing something.
> 
> Let me know, and I can start working on this.

I've noticed it with the official site on smaller screens (on components > package manager)

**Comment by kalix127** [comment]
I found the problem, and it doesn't seem like a real or actual problem.
There are 2 reasons:
1. Users are not very likely to browse docs on mobile.
2. Even if a user is on mobile, they must have a screen smaller than 290px, which is really unlikely nowadays

**Comment by Ast3risk-ops** [comment]
> I found the problem, and it doesn't seem like a real or actual problem. There are 2 reasons:
> 
>     1. Users are not very likely to browse docs on mobile.
> 
>     2. Even if a user is on mobile, they must have a screen smaller than 290px, which is really unlikely nowadays.
> 
> 
> That said, it is fixable but... is it really worth it?

Speak for yourself, plenty of people browse the *internet* on mobile. That said, it's still a really small problem.

**Comment by YasinSpace** [comment]
As you mentioned, the issue is minor and small! However, as you can see, I am on the desktop, and when I increase the submenus, they are displayed correctly in normal mode. But when the mouse hovers over them, they turn red. This hover effect has become extremely thin, and unlike the others, there's no extra space around it for aesthetic purposes. This happens even when there's some protrusion, and in the first image I sent, I was on the desktop, not on mobile! You can see that the word "Google" has clearly protruded.
![CleanShot 2025-01-05 at 13 25 32@2x](https://github.com/user-attachments/assets/8f9d6e6f-7423-489e-9def-26cee3fc8744)
VS.
![CleanShot 2025-01-05 at 13 24 01@2x](https://github.com/user-attachments/assets/9c257eb5-d645-4866-abc0-9c81ebfb8369)

**Comment by kalix127** [comment]
it makes sense.
If I have time later I'll try to fix it

**Comment by kalix127** [comment]
@YasinSpace I'm working on it now, i'll open a PR soon.
Obviously by greatly increasing the size of the sidebar, it becomes almost mandatory to remove the "toc" section on the right otherwise there is very little space left for the docs themselves.

Ignore the item names and section, I simply pasted the sections to simulate many sub-menus.
![image](https://github.com/user-attachments/assets/86cd2a19-ebef-489b-8cc1-6591316846a2)

**Comment by Ast3risk-ops** [comment]
> @YasinSpace I'm working on it now, i'll open a PR soon. Obviously by greatly increasing the size of the sidebar, it becomes almost mandatory to remove the "toc" section on the right otherwise there is very little space left for the docs themselves.
> 
> Ignore the item names and section, I simply pasted the sections to simulate many sub-menus. ![image](https://private-user-images.githubusercontent.com/127868476/400241975-86cd2a19-ebef-489b-8cc1-6591316846a2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYxMDc1ODgsIm5iZiI6MTczNjEwNzI4OCwicGF0aCI6Ii8xMjc4Njg0NzYvNDAwMjQxOTc1LTg2Y2QyYTE5LWViZWYtNDg5Yi04Y2MxLTY1OTEzMTY4NDZhMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEwNVQyMDAxMjhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03ZjQzOWQwYTRjMWFmOTQyZmU2OTYyYzc3ODEzNzdkNWI0MTYzZTc1NGE2MzI4YTZmOWI3YjAzZmYxMDBkNTU1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.JZJdJGmHXaDXHt506c4pBG_DsopLixQt7PLIVeozojo)

Which means this is more of an issue with organization than anything else. If you have that many subsections you may need to reorganize your pages. (Folders with a `_dir.yml` will appear above the pages and are better for organizing).

This should be a choice or conditional.

**Comment by kalix127** [comment]
Yes of course it's optional, that's a new prop to the aside in the app config.

**Comment by Ast3risk-ops** [comment]
> As you mentioned, the issue is minor and small! However, as you can see, I am on the desktop, and when I increase the submenus, they are displayed correctly in normal mode. But when the mouse hovers over them, they turn red. This hover effect has become extremely thin, and unlike the others, there's no extra space around it for aesthetic purposes. This happens even when there's some protrusion, and in the first image I sent, I was on the desktop, not on mobile! You can see that the word "Google Dorking" has clearly protruded.
> 
> ![CleanShot 2025-01-05 at 13 25 32@2x](https://private-user-images.githubusercontent.com/183857671/400192070-8f9d6e6f-7423-489e-9def-26cee3fc8744.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYxMDc5NDgsIm5iZiI6MTczNjEwNzY0OCwicGF0aCI6Ii8xODM4NTc2NzEvNDAwMTkyMDcwLThmOWQ2ZTZmLTc0MjMtNDg5ZS05ZGVmLTI2Y2VlM2ZjODc0NC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEwNVQyMDA3MjhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03ODQ2NTM1MjFhOTVlN2Q2MGEyNDkxZjRiZTU4ZGNhNTA3OWY2Y2Q1ZWNkNzE1NzI5NDVjODljMjM2Yzk5NmYzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.X1ARoP71UuxazqsTXekVkDFssvDuuG5hAk0Purw4jAc)
> 
> VS.
> 
> ![CleanShot 2025-01-05 at 13 24 01@2x](https://private-user-images.githubusercontent.com/183857671/400192021-9c257eb5-d645-4866-abc0-9c81ebfb8369.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYxMDc5NDgsIm5iZiI6MTczNjEwNzY0OCwicGF0aCI6Ii8xODM4NTc2NzEvNDAwMTkyMDIxLTljMjU3ZWI1LWQ2NDUtNDg2Ni1hYmMwLTljODFlYmZiODM2OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEwNVQyMDA3MjhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zODU3ZDI2ZDIyM2E1M2YyMjQzN2RmY2JmZjMzY2I0MzIxZWRiMjY2NGFmMGM3ZDVjZGQ3MDZjNWI4Y2I4YTk1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.gmNrEcjPrK64NaFqMB9fnkjiVbxsemTbI84qgYQ7G-g)

Basically I'm just saying this could be organized better (still a good idea to add the option)

**Comment by kalix127** [comment]
Do you have any suggestions?

**Comment by Ast3risk-ops** [comment]
> Do you have any suggestions?

obv. introduction should be a root-level folder.

The Methodologies section seems to be their attempt to create section index pages (which the theme should have, but doesn't) and should be reworked with less sections (i.e. do we need to specify pentesting methodologies *under* methodologies?)

**Comment by kalix127** [comment]
Having introduction as root-level is ok, but the problem remains. You have only removed 1 level of indentation no?
I hope I didn't misunderstand

**Comment by Ast3risk-ops** [comment]
> Having introduction as root-level is ok, but the problem remains. You have only removed 1 level of indentation no? I hope I didn't misunderstand

It's possible to remove more by cutting out subfolders. I explained we can cut out methodologies for pentesting as well. (methodologies too since that would be a root-level folder)

**Comment by kalix127** [comment]
Yes, I understand the meaning and I agree that it is better, but I don't have the time to do it now.
In the meantime we can use this new option, then we'll see.

**Comment by Ast3risk-ops** [comment]
> Yes, I understand the meaning and I agree that it is better, but I don't have the time to do it now. In the meantime we can use this new option, then we'll see.

You didn't make the site? The OP did.

**Comment by kalix127** [comment]
No i didn't
